bigquery/gcloud/aio/bigquery/bigquery.py

```diff
@@ -342,20 +345,46 @@ def _make_query_body(
             },
         }
 
-    async def _post_json(
-            self, url: str, body: Dict[str, Any], session: Optional[Session],
-            timeout: int) -> Dict[str, Any]:
-        payload = json.dumps(body).encode('utf-8')
+    # https://cloud.google.com/bigquery/docs/reference/rest/v2/tables/insert
+    async def create(self, table: Dict[str, Any],
+                     session: Optional[Session] = None,
+                     timeout: int = 60) -> Dict[str, Any]:
+        """Create the table specified by tableId from the dataset."""
+        project = await self.project()
+        url = (f'{API_ROOT}/projects/{project}/datasets/'
+               f'{self.dataset_name}/tables')
+        if 'tableReference' not in table:
+            table_reference = {
+                'projectId': project,
+                'datasetId': self.dataset_name,
+                'tableId': self.table_name
+            }
+            table['tableReference'] = table_reference
+
+        return await self._post_json(url, table, session, timeout)
+
+    # https://cloud.google.com/bigquery/docs/reference/rest/v2/tables/patch
+    async def patch(self, table: Dict[str, Any],
+                    session: Optional[Session] = None,
+                    timeout: int = 60) -> Dict[str, Any]:
+        """Patch an existing table specified by tableId from the dataset."""
+        project = await self.project()
+        url = (f'{API_ROOT}/projects/{project}/datasets/'
+               f'{self.dataset_name}/tables/{self.table_name}')
+        if 'tableReference' not in table:
```

Comment:
Interesting -- is there any usecase for the `tableReference` *not* matching the data you're providing here? As far as I can tell, the API docs say the `tableReference` must match the IDs listed in the URL. Since users can't modify the URL, we shouldn't let them modify the `tableReference` either (if I read that right!) since that's just a footgun waiting to happen. What do you think about overwriting `table['tableReference']` unconditionally?

Reply:
Overwriting `tableReference` makes sense, will do.

Reply:
Done.